README.rst

```diff
@@ -1,6 +1,6 @@
-.. image:: https://github.com/GalSim-developers/GalSim/workflows/GalSim%20CI/badge.svg
+.. image:: https://github.com/GalSim-developers/GalSim/workflows/GalSim%20CI/badge.svg?branch=main
```

Comment:
I also updated this link to show the badge corresponding to the run on the `main` branch, typically after merging a feature branch. I don't suppose you want to show case that tests on a development branch failed in the repo's README.